integration/container/ps_test.go

```diff
@@ -0,0 +1,79 @@
+package container
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPsFilter(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prev, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"prev",
+	)
+	require.NoError(t, err)
+
+	_, err = client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"top",
+	)
+	require.NoError(t, err)
+
+	next, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"next",
+	)
+	require.NoError(t, err)
+
+	containerID := func(containers []types.Container) []string {
+		entries := []string{}
+		for _, container := range containers {
+			entries = append(entries, container.ID)
+		}
+		return entries
+	}
+
+	f1 := filters.NewArgs()
+	f1.Add("since", "top")
+	q1, err := client.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: f1,
+	})
+	require.NoError(t, err)
+	assert.Contains(t, containerID(q1), next.ID)
```

Comment:
Can this be `Equal()` instead of `Contains()`?

Reply:
@dnephin When I run locally with:
```
TESTFLAGS="-test.run TestPsFilter" make test-integration
```
It could be `Equal()` without issues.

However, on Jenkins I noticed there are some other containers running so just used `Contains()`. As full ID is used I think it is fine.